src/App.jsx

```diff
@@ -1088,12 +1088,38 @@ const App = memo(() => {
     if (!playing) {
       video.play();
     } else {
+      let calculatedRate;
+      if (useMultiplier) {
+        switch (true) {
+          case speedMultiplier === '2x' && dir > 0:
+            calculatedRate = video.playbackRate * 2.0;
+            break;
+          case speedMultiplier === '2x' && dir < 0:
+            calculatedRate = video.playbackRate * 0.5;
+            break;
+          case speedMultiplier === '1.1x' && dir > 0:
+            calculatedRate = video.playbackRate * 1.1;
+            break;
+          case speedMultiplier === '1.1x' && dir < 0:
+            calculatedRate = video.playbackRate * 0.9091;
+            break;
+          default:
+            calculatedRate = 1.0;
+            break;
+        }
+      } else {
+        calculatedRate = video.playbackRate + (dir * 0.15);
+      }
+      // if the multiplier causes us to go faster than real time or slower than real time, stop along the way at 1.0
+      if ((calculatedRate > 1.0 && video.playbackRate < 1.0) || (calculatedRate < 1.0 && video.playbackRate > 1.0)) {
+        calculatedRate = 1.0;
+      }
       // https://github.com/mifi/lossless-cut/issues/447#issuecomment-766339083
-      const newRate = clamp(Math.round((video.playbackRate + (dir * 0.15)) * 100) / 100, 0.1, 16);
+      const newRate = clamp(Math.round(calculatedRate * 100) / 100, 0.1, 16);
```

Comment:
Is there any good reason to do rounding here? This pretty much guarantees that multiplicative speed changes won't end up where you started from (unless you cross 1.0).

I'd do the rounding in the code that displays the value. I'd also change the display to show a factor (w/ two significant digits) instead of a percentage.

Reply:
I think I added that because I wanted it to come back to 100% instead of 99%, see referenced issue https://github.com/mifi/lossless-cut/issues/447#issuecomment-766339083

Reply:
The clamping code above this already takes care of that, at least if you rewrite it the way I sketched above. 😀

In any case, when you display the value you should probably round off to nearest integer (or fraction) instead of towards zero.